src/index.ts

```diff
@@ -215,10 +215,31 @@ if (ENABLE_GRAPHQL_UPLOAD) {
 }
 
 app.use("/v2", (req, res, next) => {
-  if (req.url !== "/") return next()
+  // This matches all routes that start with '/v2'.
+  //
+  // We only want to support:
+  //   - POSTs -> requests to '/v2'
+  //   - GETs:
+  //     - requests to '/v2' (loads GraphiQL in non-prod mode)
+  //     - requests to '/v2?query=...' (runs a query)
+  if (req.method === "POST") {
+    if (req.url !== "/") return next()
+  } else if (req.method === "GET") {
+    if (req.url !== "/") {
```

Comment:
For GETs, they are allowed to be at the root. In non-prod mode this loads GraphiQL, and in prod mode this automatically provides an appropriate message about requiring a query param.